[14_1_X] Backport: Restructure and comment the tau embedding method

winterchristian opened the pull request:
#### PR description:

This is a backport for #47299 and #48408. It's similar to the backports #49290 (14_0_X), #49022 (14_2_X), #48557 (15_0_X) and #49039 (15_1_X).
This backport only serves to complete the line of backports since 14_0_X.

Because this Backport was not touched for a longer time, but is necessary for the tau embedding method, the backport of PR #50764 is also added into this PR.

#### PR validation:

Unit tests and release validations (136.901,136.902,136.903,136.904,136.905,136.906) are implemented.

cmsbuild committed:
Merge pull request #48408 from KIT-CMS/embedding_dev_CMSSW_15_1_0_pre1_improveCode

Restructure and comment the tau embedding method

(cherry picked from commit 1cf5c12622773396ce25e7565c45ba64ae21524b)

Commit:
Merge pull request #47299 from KIT-CMS/embedding_update_for_run3

Embedding update for run3

(cherry picked from commit afa07e567613f32ff2e6e13dc34e865f70259d0f)

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"337d3512aeadde47feb84f31efa35497c7e0af19":{"files":["RecoLocalCalo/Configuration/python/hcalGlobalReco_cff.py","TauAnalysis/MCEmbeddingTools/python/Merging_USER_cff.py"],"squashed":false,"time":1763548129},"53fcbe0ca413d0c846b45562de86d92263bcfc81":{"files":["Configuration/Applications/python/ConfigBuilder.py","Configuration/EventContent/python/EventContent_cff.py","Configuration/ProcessModifiers/python/tau_embedding_cleaning_cff.py","Configuration/ProcessModifiers/python/tau_embedding_emu_cff.py","Configuration/ProcessModifiers/python/tau_embedding_etauh_cff.py","Configuration/ProcessModifiers/python/tau_embedding_merging_cff.py","Configuration/ProcessModifiers/python/tau_embedding_mu_to_e_cff.py","Configuration/ProcessModifiers/python/tau_embedding_mu_to_mu_cff.py","Configuration/ProcessModifiers/python/tau_embedding_mutauh_cff.py","Configuration/ProcessModifiers/python/tau_embedding_sim_cff.py","Configuration/ProcessModifiers/python/tau_embedding_tauhtauh_cff.py","Configuration/PyReleaseValidation/python/relval_standard.py","Configuration/PyReleaseValidation/python/relval_steps.py","Configuration/StandardSequences/python/Reconstruction_cff.py","IOMC/EventVertexGenerators/python/VtxSmearedRealistic_cfi.py","PhysicsTools/NanoAOD/python/autoNANO.py","RecoLocalCalo/Configuration/python/hcalGlobalReco_cff.py","RecoLocalCalo/Configuration/python/hcalLocalRecoNZS_cff.py","RecoLocalCalo/EcalRecProducers/python/ecalPreshowerRecHit_cfi.py","RecoLocalCalo/EcalRecProducers/python/ecalRecHitGPU_cfi.py","RecoLocalCalo/EcalRecProducers/python/ecalRecHit_cfi.py","RecoLocalCalo/HcalRecProducers/python/HFPhase1Reconstructor_cfi.py","RecoLocalCalo/HcalRecProducers/python/HcalHitReconstructor_ho_cfi.py","RecoLocalMuon/CSCRecHitD/python/cscRecHitD_cfi.py","RecoLocalMuon/CSCSegment/python/cscSegments_cfi.py","RecoLocalMuon/DTRecHit/python/dt1DRecHits_LinearDriftFromDB_cfi.py","RecoLocalMuon/DTSegment/python/dt4DSegments_MTPatternReco4D_LinearDriftFromDB_cfi.py","RecoLocalMuon/RPCRecHit/python/rpcRecHits_cfi.py","RecoLocalTracker/SiStripClusterizer/python/SiStripClusterizer_RealData_cfi.py","RecoLuminosity/LumiProducer/python/bunchSpacingProducer_cfi.py","RecoTracker/IterativeTracking/python/InitialStepPreSplitting_cff.py","RecoTracker/IterativeTracking/python/InitialStep_cff.py","RecoVertex/Configuration/python/RecoVertex_cff.py","SimGeneral/MixingModule/python/SiPixelSimParameters_cfi.py","SimGeneral/MixingModule/python/SiStripSimParameters_cfi.py","SimGeneral/MixingModule/python/castorDigitizer_cfi.py","SimGeneral/MixingModule/python/ecalDigitizer_cfi.py","SimGeneral/MixingModule/python/hcalDigitizer_cfi.py","TauAnalysis/MCEmbeddingTools/docs/README","TauAnalysis/MCEmbeddingTools/docs/README.md","TauAnalysis/MCEmbeddingTools/docs/cmsDriver_cmds_DATA_RUN3.sh","TauAnalysis/MCEmbeddingTools/interface/CaloCleaner.h","TauAnalysis/MCEmbeddingTools/interface/CollectionMerger.h","TauAnalysis/MCEmbeddingTools/interface/DoubleCollectionMerger.h","TauAnalysis/MCEmbeddingTools/interface/EmbeddingBeamSpotOnlineProducer.h","TauAnalysis/MCEmbeddingTools/interface/MuonDetCleaner.h","TauAnalysis/MCEmbeddingTools/interface/TrackMergeremb.h","TauAnalysis/MCEmbeddingTools/interface/TrackerCleaner.h","TauAnalysis/MCEmbeddingTools/plugins/CaloCleaner.cc","TauAnalysis/MCEmbeddingTools/plugins/CollectionMerger.cc","TauAnalysis/MCEmbeddingTools/plugins/DoubleCollectionMerger.cc","TauAnalysis/MCEmbeddingTools/plugins/EmbeddingBeamSpotOnlineProducer.cc","TauAnalysis/MCEmbeddingTools/plugins/EmbeddingLHEProducer.cc","TauAnalysis/MCEmbeddingTools/plugins/MuMuForEmbeddingSelector.cc","TauAnalysis/MCEmbeddingTools/plugins/MuonDetCleaner.cc","TauAnalysis/MCEmbeddingTools/plugins/TrackMergeremb.cc","TauAnalysis/MCEmbeddingTools/plugins/TrackerCleaner.cc","TauAnalysis/MCEmbeddingTools/python/Cleaning_RECO_cff.py","TauAnalysis/MCEmbeddingTools/python/EmbeddingBeamSpotOnline_cfi.py","TauAnalysis/MCEmbeddingTools/python/EmbeddingHltPixelVerticesProducer_cfi.py","TauAnalysis/MCEmbeddingTools/python/EmbeddingLHEProducer_cfi.py","TauAnalysis/MCEmbeddingTools/python/EmbeddingPythia8Hadronizer_cfi.py","TauAnalysis/MCEmbeddingTools/python/EmbeddingVertexCorrector_cfi.py","TauAnalysis/MCEmbeddingTools/python/Embedding_EventContent_cff.py","TauAnalysis/MCEmbeddingTools/python/LHE_USER_cff.py","TauAnalysis/MCEmbeddingTools/python/Merging_USER_cff.py","TauAnalysis/MCEmbeddingTools/python/Nano_cff.py","TauAnalysis/MCEmbeddingTools/python/Selection_FILTER_cff.py","TauAnalysis/MCEmbeddingTools/python/Simulation_GEN_cfi.py","TauAnalysis/MCEmbeddingTools/python/Simulation_HLT_customiser_cff.py","TauAnalysis/MCEmbeddingTools/python/Simulation_RECO_cff.py","TauAnalysis/MCEmbeddingTools/python/customisers.py","TauAnalysis/MCEmbeddingTools/test/run_2016postVFPUL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2016preVFPUL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2017UL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2018UL_workflow_tests.sh","TauAnalysis/MCEmbeddingTools/test/run_2022_workflow_tests.sh"],"squashed":false,"time":1763487594},"71ac4817ee1f403314db6a20df6f2b200685cea7":{"files":["PhysicsTools/NanoAOD/python/triggerObjects_cff.py","PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py","TauAnalysis/MCEmbeddingTools/python/Merging_USER_cff.py","TauAnalysis/MCEmbeddingTools/python/Simulation_GEN_cfi.py"],"squashed":false,"time":1776689542},"ec9cc8f4a52ce505601eff5b6f567e5fdf350f9d":{"files":["TauAnalysis/MCEmbeddingTools/python/SelectingProcedure_cff.py","TauAnalysis/MCEmbeddingTools/python/customisers.py","TauAnalysis/MCEmbeddingTools/test/BuildFile.xml","TauAnalysis/MCEmbeddingTools/test/run_2022_workflow_tests.sh"],"squashed":false,"time":1763484626}},"emoji":{"3551182214":"-1","3551184076":"+1","3551184232":"+1","3552450899":"+1","3553126496":"+1","3553460972":"+1","3571404080":"+1","3571404636":"+1","3572101553":"+1","3669063893":"+1","3669064203":"+1","4295560547":"+1","4296684207":"+1","4331026283":"+1","4334674479":"+1","4343843892":"+1"},"last_seen_sha":"71ac4817ee1f403314db6a20df6f2b200685cea7","signatures":{"3553126496":"337d3512aeadde47feb84f31efa35497c7e0af19","3553460972":"337d3512aeadde47feb84f31efa35497c7e0af19","3572101553":"337d3512aeadde47feb84f31efa35497c7e0af19","3669064203":"337d3512aeadde47feb84f31efa35497c7e0af19","4296684207":"71ac4817ee1f403314db6a20df6f2b200685cea7","4331026283":"71ac4817ee1f403314db6a20df6f2b200685cea7","4334674479":"71ac4817ee1f403314db6a20df6f2b200685cea7","4343843892":"71ac4817ee1f403314db6a20df6f2b200685cea7"}} -->

AdrianoDee commented:
please test workflows 136.901,136.902,136.903,136.904,136.905,136.906)

AdrianoDee commented:
code-checks

Commit:
Backport fixes due to cherry picking future commits

cmsbuild commented:
+1

**Size**: This PR adds an extra 1856KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5aaad/49547/summary.html
**COMMIT**: 337d3512aeadde47feb84f31efa35497c7e0af19
**CMSSW**: CMSSW_14_1_X_2025-11-16-0000/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/49410/49547/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 3 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2025-11-16-0000+d5aaad/71850/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 16 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2025-11-16-0000+d5aaad/71850/validateJR.html)
 - DQMHistoTests: Total files compared: 47
 - DQMHistoTests: Total histograms compared: 3672688
 - DQMHistoTests: [Total failures: 15](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2025-11-16-0000+d5aaad/71850/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3672653
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 46 files compared)
 - Checked 203 log files, 174 edm output root files, 47 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2025-11-16-0000+d5aaad/71850/triggerResults): no differences found

AdrianoDee commented:
please test

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_14_1_X IBs (tests are also fine) and once validation in the development release cycle CMSSW_16_0_X is complete. This pull request will now be reviewed by the release team before it's merged. @mandrenguyen, @ftenchini, @sextonkennedy (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Size**: This PR adds an extra 12KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5aaad/50330/summary.html
**COMMIT**: 337d3512aeadde47feb84f31efa35497c7e0af19
**CMSSW**: CMSSW_14_1_X_2025-12-14-0000/el8_amd64_gcc12
**Additional Tests**: NANO
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/49410/50330/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Failed Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test test-das-selected-lumis had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially [removed 2 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2025-12-14-0000+d5aaad/72432/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 6 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2025-12-14-0000+d5aaad/72432/validateJR.html)
 - DQMHistoTests: Total files compared: 47
 - DQMHistoTests: Total histograms compared: 3672688
 - DQMHistoTests: [Total failures: 6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2025-12-14-0000+d5aaad/72432/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3672662
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 46 files compared)
 - Checked 203 log files, 174 edm output root files, 47 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2025-12-14-0000+d5aaad/72432/triggerResults): no differences found

**Nano size comparison Summary**:

| Sample | kb/ev | ref kb/ev | diff kb/ev | ev/s/thd | ref ev/s/thd | diff rate | mem/thd | ref mem/thd |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 2500.001_NANOmc106Xul16v2 | 2.791 | 2.791 | 0.000 ( +0.0% ) | 7.63 | 6.04 | +26.4% | 2.143 | 2.315 |
| 2500.002_NANOmc106Xul17v2 | 2.903 | 2.903 | 0.000 ( +0.0% ) | 6.78 | 5.50 | +23.3% | 2.566 | 2.748 |
| 2500.003_NANOmc106Xul18v2 | 2.851 | 2.851 | 0.000 ( +0.0% ) | 7.06 | 5.67 | +24.7% | 2.565 | 2.729 |
| 2500.011_NANOdata106Xul16v2 | 1.456 | 1.456 | 0.000 ( +0.0% ) | 13.18 | 10.43 | +26.3% | 2.258 | 2.408 |
| 2500.012_NANOdata106Xul17v2 | 1.914 | 1.914 | 0.000 ( +0.0% ) | 7.17 | 5.75 | +24.8% | 2.398 | 2.607 |
| 2500.013_NANOdata106Xul18v2 | 1.770 | 1.770 | 0.000 ( +0.0% ) | 10.54 | 8.18 | +28.9% | 2.344 | 2.511 |
| 2500.021_NANOmcUL16APVreMINI | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.97 | 1.48 | +32.8% | 2.276 | 2.480 |
| 2500.022_NANOmcUL16reMINI | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.92 | 1.57 | +22.2% | 2.267 | 2.478 |
| 2500.023_NANOmcUL17reMINI | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.99 | 1.48 | +34.3% | 2.212 | 2.346 |
| 2500.024_NANOmcUL18reMINI | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.53 | 1.13 | +35.7% | 2.382 | 2.578 |
| 2500.031_NANOdataUL16APVreMINI | 0.035 | 0.035 | 0.000 ( +0.0% ) | 1.80 | 1.35 | +33.5% | 2.374 | 2.567 |
| 2500.032_NANOdataUL16reMINI | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.93 | 1.39 | +39.0% | 2.304 | 2.508 |
| 2500.033_NANOdataUL17reMINI | 0.037 | 0.037 | -0.000 ( -0.1% ) | 1.72 | 1.30 | +31.9% | 2.410 | 2.596 |
| 2500.034_NANOdataUL18reMINI | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.74 | 1.30 | +34.3% | 2.392 | 2.589 |
| 2500.101_NANOmc130X | 2.643 | 2.643 | 0.000 ( +0.0% ) | 21.77 | 16.08 | +35.4% | 2.301 | 2.482 |
| 2500.111_NANOdata130Xrun3 | 1.339 | 1.339 | 0.000 ( +0.0% ) | 49.46 | 33.87 | +46.0% | 2.036 | 2.188 |
| 2500.112_NANOdata130Xrun3 | 1.745 | 1.745 | 0.000 ( +0.0% ) | 36.30 | 26.17 | +38.7% | 2.077 | 2.280 |
| 2500.131_ScoutingNANOdata130Xrun3 | 0.747 | 0.747 | 0.000 ( +0.0% ) | 37.24 | 29.78 | +25.1% | 1.365 | 1.502 |
| 2500.201_NANOmc140X | 2.478 | 2.478 | 0.000 ( +0.0% ) | 17.77 | 13.18 | +34.8% | 1.928 | 2.057 |
| 2500.211_NANOdata140Xrun3 | 1.592 | 1.592 | 0.000 ( +0.0% ) | 44.33 | 30.29 | +46.4% | 2.060 | 2.276 |
| 2500.212_NANOdata140Xrun3 | 2.033 | 2.033 | 0.000 ( +0.0% ) | 32.75 | 19.77 | +65.7% | 2.100 | 2.360 |
| 2500.221_muPOGNANOmc140X | 2.006 | 2.006 | 0.000 ( +0.0% ) | 18.72 | 14.03 | +33.4% | 1.774 | 1.985 |
| 2500.222_EGMNANOmc140X | 3.140 | 3.140 | 0.000 ( +0.0% ) | 17.39 | 13.40 | +29.7% | 1.837 | 2.072 |
| 2500.223_BTVNANOmc140X | 8.888 | 8.888 | 0.000 ( +0.0% ) | 6.82 | 5.22 | +30.7% | 1.864 | 2.099 |
| 2500.224_jmeNANOmc140X | 5.515 | 5.515 | 0.000 ( +0.0% ) | 2.74 | 2.20 | +24.9% | 1.904 | 2.125 |
| 2500.225_jmeNANOrePuppimc140X | 5.533 | 5.533 | 0.000 ( +0.0% ) | 2.47 | 1.97 | +25.6% | 1.916 | 2.152 |
| 2500.226_lepTrackInfoNANOmc140X | 2.973 | 2.973 | 0.000 ( +0.0% ) | 18.31 | 13.60 | +34.6% | 1.832 | 2.062 |
| 2500.227_ScoutingNANOmc140X | 1.437 | 1.437 | 0.000 ( +0.0% ) | 24.05 | 19.53 | +23.1% | 1.307 | 1.429 |
| 2500.231_muPOGNANO140Xrun3 | 1.407 | 1.407 | 0.000 ( +0.0% ) | 33.74 | 22.62 | +49.1% | 1.937 | 2.171 |
| 2500.232_EGMNANOdata140Xrun3 | 2.199 | 2.199 | 0.000 ( +0.0% ) | 30.40 | 23.11 | +31.6% | 2.005 | 2.269 |
| 2500.233_BTVNANOdata140Xrun3 | 4.671 | 4.671 | 0.000 ( +0.0% ) | 11.66 | 9.02 | +29.3% | 2.015 | 2.276 |
| 2500.234_jmeNANOdata140Xrun3 | 3.307 | 3.307 | 0.000 ( +0.0% ) | 3.79 | 3.10 | +22.3% | 2.044 | 2.302 |
| 2500.235_jmeNANOrePuppidata140Xrun3 | 3.317 | 3.317 | 0.000 ( +0.0% ) | 3.45 | 2.79 | +23.8% | 2.086 | 2.323 |
| 2500.236_lepTrackInfoNANOdata140Xrun3 | 2.085 | 2.085 | 0.000 ( +0.0% ) | 32.72 | 23.91 | +36.8% | 1.999 | 2.259 |
| 2500.237_ScoutingNANOdata140Xrun3 | 1.016 | 1.016 | 0.000 ( +0.0% ) | 35.45 | 29.07 | +21.9% | 1.322 | 1.456 |
| 2500.241_l1DPGNANO140Xrun3 | 9.404 | 9.404 | 0.000 ( +0.0% ) | 7.30 | 5.11 | +42.9% | 1.772 | 1.917 |
| 2500.242_muDPGNANO140Xrun3 | 10.331 | 10.331 | 0.000 ( +0.0% ) | 1.64 | 1.34 | +22.8% | 1.645 | 1.697 |
| 2500.243_muDPGNANOBkg140Xrun3 | 2.712 | 2.712 | 0.000 ( +0.0% ) | 15.64 | 11.15 | +40.2% | 1.003 | 1.060 |
| 2500.244_hcalDPGNANO140Xrun3 | 485.976 | 485.976 | 0.000 ( +0.0% ) | 1.14 | 0.96 | +19.0% | 1.569 | 1.671 |
| 2500.245_hcalDPGCalibNANO140Xrun3 | 823.202 | 823.202 | 0.000 ( +0.0% ) | 1.52 | 1.20 | +26.6% | 1.542 | 1.680 |
| 2500.901_TTbarMINIAOD10.6_UL18v2 | 1.749 | 1.749 | 0.000 ( +0.0% ) | 48.32 | 32.43 | +49.0% | 1.334 | 1.405 |
| 2500.902_TTbarMINIAOD14.0 | 1.598 | 1.598 | 0.000 ( +0.0% ) | 50.78 | 40.02 | +26.9% | 1.285 | 1.310 |
| 2500.911_DYToLL_M-50_13TeV_pythia8 | 13.931 | 13.931 | 0.000 ( +0.0% ) | 7.45 | 4.78 | +55.8% | 1.023 | 1.074 |
| 2500.912_DYToll01234Jets_5f_LO_MLM_Madgraph_LHE_13TeV | 0.729 | 0.437 | 0.293 ( +67.1% ) | 2.74 | 3.27 | -16.4% | 0.965 | 0.965 |
| 2500.913_TTbar_Pow_LHE_13TeV | 0.110 | 0.110 | 0.000 ( +0.0% ) | 2.58 | 2.21 | +16.9% | 0.976 | 0.972 |


## NANO Comparison Summary



**Summary**:
 - You potentially [added 29 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_1_X_2025-12-14-0000+d5aaad/72429/validateJR/logRootQA-events.log) to the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_1_X_2025-12-14-0000+d5aaad/72429/validateJR.html)
 - DQMHistoTests: Total files compared: 21
 - DQMHistoTests: Total histograms compared: 54921
 - DQMHistoTests: [Total failures: 0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_1_X_2025-12-14-0000+d5aaad/72429/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 54921
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 20 files compared)
 - Checked 102 log files, 58 edm output root files, 21 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_1_X_2025-12-14-0000+d5aaad/72429/triggerResults): no differences found

winterchristian committed:
Backport of #50764 Bug Fix Tau Embedding HLT info (#7)

Contains the following commits:
5999474 debug embedding hlt by adding modifier rules to use embedding hlt collections
ffea69a disable Final State Radiation for all embedding final states

Co-authored-by: Christian Winter <c.winter@kit.edu>

cmsbuild commented:
Pull request #49410 was updated. @Moanwar, @battibass, @civanch, @cmsbuild, @davidlange6, @fabiocos, @ftenchini, @ftorrresd, @jfernan2, @kpedro88, @mandrenguyen, @mdhildreth, @srimanob can you please check and sign again.

jfernan2 commented:
This PR is declared in https://github.com/cms-sw/cmssw/pull/50764 as a backport but the content is totally different since this PR is old. Could you please provide a 14_1_X backport?

jfernan2 commented:
@winterchristian this PR does not seem like a current backport of #50764 
Can you check please?

winterchristian commented:
Hi @jfernan2, your right. The problem is, that the PR #50764 needs the changes of the PR #47299 and #48408. Those two PRs are backported in this PR.
So I can't provide you with a backport of #50764 without shipping the changes of this backport in it.
And because this backport already exists I just merged the changes of the PR #50764 into this backport, so it can be merged together into `CMSSW_14_1_X`

jfernan2 commented:
please test
OK, let's refresh the test

cmsbuild commented:
+1

**Size**: This PR adds an extra 12KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5aaad/52817/summary.html
**COMMIT**: 71ac4817ee1f403314db6a20df6f2b200685cea7
**CMSSW**: CMSSW_14_1_X_2026-04-19-0000/el8_amd64_gcc12
**Additional Tests**: NANO
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/49410/52817/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 6 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2026-04-19-0000+d5aaad/74817/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 89 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2026-04-19-0000+d5aaad/74817/validateJR.html)
 - DQMHistoTests: Total files compared: 47
 - DQMHistoTests: Total histograms compared: 3672688
 - DQMHistoTests: [Total failures: 3429](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2026-04-19-0000+d5aaad/74817/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3669239
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 46 files compared)
 - Checked 203 log files, 174 edm output root files, 47 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2026-04-19-0000+d5aaad/74817/triggerResults): no differences found

**Nano size comparison Summary**:

| Sample | kb/ev | ref kb/ev | diff kb/ev | ev/s/thd | ref ev/s/thd | diff rate | mem/thd | ref mem/thd |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 2500.001_NANOmc106Xul16v2 | 2.791 | 2.791 | 0.000 ( +0.0% ) | 7.42 | 6.91 | +7.4% | 2.312 | 2.234 |
| 2500.002_NANOmc106Xul17v2 | 2.903 | 2.903 | 0.000 ( +0.0% ) | 6.63 | 6.16 | +7.6% | 2.736 | 2.671 |
| 2500.003_NANOmc106Xul18v2 | 2.851 | 2.851 | 0.000 ( +0.0% ) | 6.90 | 6.37 | +8.4% | 2.725 | 2.674 |
| 2500.011_NANOdata106Xul16v2 | 1.456 | 1.456 | 0.000 ( +0.0% ) | 12.78 | 11.75 | +8.8% | 2.363 | 2.347 |
| 2500.012_NANOdata106Xul17v2 | 1.914 | 1.914 | 0.000 ( +0.0% ) | 7.02 | 6.59 | +6.5% | 2.507 | 2.505 |
| 2500.013_NANOdata106Xul18v2 | 1.770 | 1.770 | 0.000 ( +0.0% ) | 10.07 | 9.39 | +7.2% | 2.418 | 2.412 |
| 2500.021_NANOmcUL16APVreMINI | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.94 | 1.63 | +19.4% | 2.299 | 2.290 |
| 2500.022_NANOmcUL16reMINI | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.86 | 1.63 | +14.4% | 2.304 | 2.288 |
| 2500.023_NANOmcUL17reMINI | 0.022 | 0.022 | -0.000 ( -0.2% ) | 1.80 | 1.53 | +17.2% | 2.255 | 2.251 |
| 2500.024_NANOmcUL18reMINI | 0.022 | 0.022 | 0.000 ( +0.0% ) | 1.46 | 1.20 | +21.4% | 2.424 | 2.418 |
| 2500.031_NANOdataUL16APVreMINI | 0.035 | 0.035 | 0.000 ( +0.0% ) | 1.76 | 1.40 | +25.5% | 2.408 | 2.400 |
| 2500.032_NANOdataUL16reMINI | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.80 | 1.51 | +18.9% | 2.352 | 2.328 |
| 2500.033_NANOdataUL17reMINI | 0.037 | 0.037 | 0.000 ( +0.1% ) | 1.59 | 1.30 | +22.0% | 2.451 | 2.458 |
| 2500.034_NANOdataUL18reMINI | 0.036 | 0.036 | 0.000 ( +0.0% ) | 1.65 | 1.35 | +22.6% | 2.434 | 2.424 |
| 2500.101_NANOmc130X | 2.643 | 2.643 | 0.000 ( +0.0% ) | 20.22 | 17.98 | +12.4% | 2.436 | 2.436 |
| 2500.111_NANOdata130Xrun3 | 1.339 | 1.339 | 0.000 ( +0.0% ) | 43.71 | 37.58 | +16.3% | 2.190 | 2.187 |
| 2500.112_NANOdata130Xrun3 | 1.745 | 1.745 | 0.000 ( +0.0% ) | 28.28 | 28.31 | -0.1% | 2.113 | 2.109 |
| 2500.131_ScoutingNANOdata130Xrun3 | 0.747 | 0.747 | 0.000 ( +0.0% ) | 35.62 | 32.55 | +9.4% | 1.506 | 1.435 |
| 2500.201_NANOmc140X | 2.478 | 2.478 | 0.000 ( +0.0% ) | 16.12 | 15.07 | +6.9% | 1.970 | 1.967 |
| 2500.211_NANOdata140Xrun3 | 1.592 | 1.592 | 0.000 ( +0.0% ) | 36.83 | 34.57 | +6.5% | 2.095 | 2.095 |
| 2500.212_NANOdata140Xrun3 | 2.033 | 2.033 | 0.000 ( +0.0% ) | 30.47 | 25.25 | +20.7% | 2.131 | 2.160 |
| 2500.221_muPOGNANOmc140X | 2.006 | 2.006 | 0.000 ( +0.0% ) | 17.80 | 15.63 | +13.9% | 1.813 | 1.808 |
| 2500.222_EGMNANOmc140X | 3.140 | 3.140 | 0.000 ( +0.0% ) | 16.42 | 14.19 | +15.7% | 1.875 | 1.864 |
| 2500.223_BTVNANOmc140X | 8.888 | 8.888 | 0.000 ( +0.0% ) | 6.49 | 5.67 | +14.3% | 1.875 | 1.866 |
| 2500.224_jmeNANOmc140X | 5.515 | 5.515 | 0.000 ( +0.0% ) | 2.65 | 2.39 | +10.8% | 1.976 | 2.120 |
| 2500.225_jmeNANOrePuppimc140X | 5.533 | 5.533 | 0.000 ( +0.0% ) | 2.41 | 2.23 | +7.7% | 1.992 | 1.959 |
| 2500.226_lepTrackInfoNANOmc140X | 2.973 | 2.973 | 0.000 ( +0.0% ) | 17.23 | 15.53 | +10.9% | 1.871 | 1.863 |
| 2500.227_ScoutingNANOmc140X | 1.437 | 1.437 | 0.000 ( +0.0% ) | 23.08 | 21.30 | +8.4% | 1.346 | 1.347 |
| 2500.231_muPOGNANO140Xrun3 | 1.407 | 1.407 | 0.000 ( +0.0% ) | 30.43 | 26.22 | +16.0% | 1.949 | 1.948 |
| 2500.232_EGMNANOdata140Xrun3 | 2.199 | 2.199 | 0.000 ( +0.0% ) | 27.36 | 24.58 | +11.3% | 2.035 | 2.028 |
| 2500.233_BTVNANOdata140Xrun3 | 4.671 | 4.671 | 0.000 ( +0.0% ) | 11.00 | 10.01 | +10.0% | 2.049 | 2.047 |
| 2500.234_jmeNANOdata140Xrun3 | 3.307 | 3.307 | 0.000 ( +0.0% ) | 3.68 | 3.34 | +10.1% | 2.064 | 2.065 |
| 2500.235_jmeNANOrePuppidata140Xrun3 | 3.317 | 3.317 | 0.000 ( +0.0% ) | 3.32 | 3.06 | +8.4% | 2.117 | 2.114 |
| 2500.236_lepTrackInfoNANOdata140Xrun3 | 2.085 | 2.085 | 0.000 ( +0.0% ) | 30.48 | 26.45 | +15.2% | 2.042 | 2.045 |
| 2500.237_ScoutingNANOdata140Xrun3 | 1.016 | 1.016 | 0.000 ( +0.0% ) | 32.16 | 32.12 | +0.1% | 1.408 | 1.418 |
| 2500.241_l1DPGNANO140Xrun3 | 9.404 | 9.404 | 0.000 ( +0.0% ) | 7.22 | 5.94 | +21.5% | 1.776 | 1.786 |
| 2500.242_muDPGNANO140Xrun3 | 10.331 | 10.331 | 0.000 ( +0.0% ) | 1.46 | 1.45 | +0.3% | 1.698 | 1.696 |
| 2500.243_muDPGNANOBkg140Xrun3 | 2.712 | 2.712 | 0.000 ( +0.0% ) | 15.63 | 13.84 | +12.9% | 1.065 | 1.062 |
| 2500.244_hcalDPGNANO140Xrun3 | 485.976 | 485.976 | 0.000 ( +0.0% ) | 1.12 | 1.07 | +4.9% | 1.598 | 1.575 |
| 2500.245_hcalDPGCalibNANO140Xrun3 | 823.202 | 823.202 | 0.000 ( +0.0% ) | 1.49 | 1.38 | +7.9% | 1.577 | 1.552 |
| 2500.901_TTbarMINIAOD10.6_UL18v2 | 1.749 | 1.749 | 0.000 ( +0.0% ) | 45.13 | 44.74 | +0.9% | 1.410 | 1.407 |
| 2500.902_TTbarMINIAOD14.0 | 1.598 | 1.598 | 0.000 ( +0.0% ) | 46.37 | 47.51 | -2.4% | 1.315 | 1.311 |
| 2500.911_DYToLL_M-50_13TeV_pythia8 | 13.931 | 13.931 | 0.000 ( +0.0% ) | 6.13 | 6.88 | -10.9% | 1.079 | 1.075 |
| 2500.912_DYToll01234Jets_5f_LO_MLM_Madgraph_LHE_13TeV | 0.199 | 0.150 | 0.049 ( +32.8% ) | 2.75 | 3.41 | -19.4% | 0.971 | 0.968 |
| 2500.913_TTbar_Pow_LHE_13TeV | 0.110 | 0.110 | 0.000 ( +0.0% ) | 2.57 | 2.25 | +14.0% | 0.977 | 0.969 |


## NANO Comparison Summary



**Summary**:
 - You potentially [removed 229 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_1_X_2026-04-19-0000+d5aaad/74816/validateJR/logRootQA-events.log) from the logs
 - ROOTFileChecks: Some differences in event products or their sizes found
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_1_X_2026-04-19-0000+d5aaad/74816/validateJR.html)
 - DQMHistoTests: Total files compared: 21
 - DQMHistoTests: Total histograms compared: 54921
 - DQMHistoTests: [Total failures: 0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_1_X_2026-04-19-0000+d5aaad/74816/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 54921
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 20 files compared)
 - Checked 102 log files, 58 edm output root files, 21 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisonsNANO/CMSSW_14_1_X_2026-04-19-0000+d5aaad/74816/triggerResults): no differences found


## Max Memory Comparisons exceeding threshold 

@cms-sw/core-l2 , I found 1 workflow step(s) with memory usage exceeding the error threshold:
<details>
<summary>Expand to see workflows ...</summary>

- Error: Workflow 332.0_Pyquen_ZeemumuJets_pt10_5362GeV_2024 step2 max memory diff -334.1 exceeds +/- 90.0 MiB 
</details>

jfernan2 commented:
+1

ftorrresd commented:
+1

civanch commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_14_1_X IBs (tests are also fine) and once validation in the development release cycle CMSSW_17_0_X is complete. This pull request will now be reviewed by the release team before it's merged. @mandrenguyen, @ftenchini, @sextonkennedy (and backports should be raised in the release meeting by the corresponding L2)

mandrenguyen commented:
+1
